    # Publish joint states
    joint_state_controller:
        type: joint_state_controller/JointStateController
        publish_rate: 100

    position_joint_trajectory_controller:
        type: position_controllers/JointTrajectoryController
        joints:
            - panda_joint1
            - panda_joint2
            - panda_joint3
            - panda_joint4
            - panda_joint5
            - panda_joint6
            - panda_joint7

        constraints:
            goal_time: 2.0

        state_publish_rate: 100

    panda_hand_controller:
        type: position_controllers/JointTrajectoryController
        joints:
            - panda_finger_joint1
            - panda_finger_joint2

        state_publish_rate: 25
    joint_position_controller:
        type: panda_simulation/JointPositionController
        arm_id: panda
        joint_names:
            - panda_joint1
            - panda_joint2
            - panda_joint3
            - panda_joint4
            - panda_joint5
            - panda_joint6
            - panda_joint7
        gains: [
            1, 1, 1, 1, 1, 1, 1
        ]